lib/remoteproc/remoteproc_virtio.c

```diff
@@ -16,6 +16,140 @@
 #include <metal/utilities.h>
 #include <metal/alloc.h>
 
+static void rproc_virtio_delete_virtqueues(struct virtio_device *vdev)
+{
+	struct virtio_vring_info *vring_info;
+	unsigned int i;
+
+	if (vdev->vrings_info != NULL) {
+		for (i = 0; i < vdev->vrings_num; i++) {
+			vring_info = &vdev->vrings_info[i];
+			if (vring_info->vq != NULL) {
+				virtqueue_free(vring_info->vq);
+			}
+		}
+
+		metal_free_memory(vdev->vrings_info);
+	}
+}
+
+static int rproc_virtio_create_virtqueue(struct virtio_device *vdev,
+					 unsigned int flags,
+					 unsigned int i,
+					 const char *name,
+					 vq_callback callback)
+{
+	struct remoteproc_virtio *rpvdev;
+	struct fw_rsc_vdev_vring *vring_rsc;
+	struct fw_rsc_vdev *vdev_rsc;
+	struct remoteproc *rproc;
```

Comment:
unexpected dependency with remoteproc.c ( this structure does not exist for some projects  implementing device part)

Reply:
Done